src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -159,16 +168,35 @@ private BoundExpression VisitCollectionBuilderCollectionExpression(BoundCollecti
             Debug.Assert(!_inExpressionLambda);
             Debug.Assert(node.Type is { });
 
+            var syntax = node.Syntax;
+            var elements = node.Elements;
             var constructMethod = node.CollectionBuilderMethod;
+
             Debug.Assert(constructMethod is { });
             Debug.Assert(constructMethod.ReturnType.Equals(node.Type, TypeCompareKind.AllIgnoreOptions));
 
             var spanType = (NamedTypeSymbol)constructMethod.Parameters[0].Type;
             Debug.Assert(spanType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions));
 
-            var span = VisitArrayOrSpanCollectionExpression(node, spanType, spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0].Type);
-            return new BoundCall(
-                node.Syntax,
+            var elementType = spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0];
+            var locals = ArrayBuilder<LocalSymbol>.GetInstance();
+            var sideEffects = ArrayBuilder<BoundExpression>.GetInstance();
+            BoundExpression span;
+
+            if (elements.Length > 0
+                && !elements.Any(i => i is BoundCollectionExpressionSpreadElement)
```

Comment:
I am a bit worried, from a usability perspective, about the potential for a perf cliff if you include a spread, and whether there's going to be advice of "yeah, for perf you shouldn't use spreads". Think we may want to push on the "Give me a span, on the stack if possible" API for allowing us to make this more of a gradient.